### DISCUSSION TITLE
Hidden message reply

### DISCUSSION THREAD

**Discussion by Darth-Coin** [stackernews/stacker.news#3008] [Ideas]
**A reply feature that will hide the text for other readers**
Will be nice and useful to have a way to reply to somebody, but only that user could see the text.
Other stackers MUST pay a fee to see it.
The fee could be defined by the OP when is replying.

This could be very useful when stackers are talking about some more private information but still want to share it with others.
Could have many use cases on SN.

Please consider it.

**Comment by huumn** [comment]
I like the private reply idea! I'm not sure about people paying to see it, but I definitely like private messages!

**Comment by ekzyis** [comment]
> but I definitely like private messages!

you mean in a different way than just #56 ?

**Comment by Darth-Coin** [comment]
#56 is something different.
Hidden message is a public message with the option to be hidden for the rest but also option of pay to view.
The fee could go to SN pool just fine.

**Comment by ekzyis** [comment]
Yes, I understood, was just wondering in what way this here would be different from #56 without the paywall (which I am also not sure about).

Basically: Isn't this #56 just with a paywall? So if we remove the paywall, then it's #56?

> This could be very useful when stackers are talking about some more private information but still want to share it with others.

Can you share some examples? This sound like an oxymoron for me. If something is truly private, I don't want other people to see it even if they pay for it

**Comment by huumn** [comment]
> If something is truly private, I don't want other people to see it even if they pay for it

Same.

> you mean in a different way than just https://github.com/stackernews/stacker.news/issues/56 ?

Yes, because a private reply inherits the context of the thread. In a way, it's more natural and fits within existing ergonomics. A DM is like it's own thing. A metaphor might help.

- A private reply is like pulling you aside at a party to say something private about what's going on at the party, like "hey your pant zipper is down" or "hey that guy is a total scammer, don't trust him"
    - I can do this in a DM, but it may be better UX if it just showed up *inside* of the relevant thread

- A DM is like going out to dinner with you

It'd be cool if we can design a DM/private reply system that is somehow a hybrid of these ideas rather than independent things, but I'm not sure.

**Comment by ekzyis** [comment]
> Yes, because a private reply inherits the context of the thread. In a way, it's more natural and fits within existing ergonomics. A DM is like it's own thing.

I see, so something related to #350

**Comment by huumn** [comment]
Somewhat, but still not the same. I updated my comment. 

These are 3 different problems and 3 different solutions. It'd be nice to solve them all with 1 solution but it might not be possible.

**Comment by Darth-Coin** [comment]
Maybe I want to send a PR-13 meme in a reply.
Like k00b said to me that is not nice to post them.
So I could hide them behind a paywall for example.

Maybe I want to send a reply with some text that is dedicated and related to previous comments, but want to be shown first only to the OP. If others really want to see it, must pay to SN pool. Is more like a game let's say.

**Comment by huumn** [comment]
Added blocked label because we want private replies to be E2EE.

**Comment by Darth-Coin** [comment]
Private replies / messages are something else.
What I wanted to say is like a spoiler alert, the comment in a public post is blurred and only those that pay to view will be able to see the spoiler.

Think about also about this scenario - I want to hide in a "spoiler alert" blurred comment some information that can be visible:
- to only those who pay a minimum zap, defined by me
- to all stackers, IF one of them pay the maximum fee to reveal the whole message to the whole SN.

Like a bounty but the way around :)

Example - some stacker will ask something specific about how to do X thing.
So I will reply with a comment, but I will put a bounty on revealing it because the information in it, worth sats. So I define min and max. Each stacker if they pay min fee, will reveal the comment only for them. If a stacker pays the max fee, it will reveal to all stackers. 

If more stackers paid the min fee, but in total was reaching the max fee, then the message is revealed to all readers.

This feature could be used in so many ways and open a lot of SN "games" that could incentivize users for good answers.
